wa/workloads/speedometer/__init__.py

```diff
@@ -12,57 +12,295 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+from http.server import SimpleHTTPRequestHandler, HTTPServer
+import logging
 import os
 import re
+import subprocess
+import tarfile
+import threading
+import time
 
-from wa import ApkUiautoWorkload, Parameter
-from wa.framework.exception import ValidationError, WorkloadError
-from wa.utils.types import list_of_strs
-from wa.utils.misc import unique
+from wa import Parameter, Workload
+from wa.framework.exception import WorkloadError
 
+from devlib.utils.android import adb_command
 
-class Speedometer(ApkUiautoWorkload):
 
-    name = 'speedometer'
-    package_names = ['com.android.chrome']
-    regex = re.compile(r'Speedometer Score ([\d.]+)')
-    versions = ['1.0', '2.0']
-    description = '''
-    A workload to execute the speedometer web based benchmark
+class ArchiveServer:
+    def __init__(self):
+        self._port = None
+
+    def start(self, document_root, target):
+        class _DifferentDirectoryHTTPRequestHandler(SimpleHTTPRequestHandler):
+            """A version of SimpleHTTPRequestHandler that allows us to serve
+            relative files from a different directory than the current one.
+            This directory is captured in |document_root|. It also suppresses
+            logging."""
+
+            def translate_path(self, path):
+                path = SimpleHTTPRequestHandler.translate_path(self, path)
+                requested_uri = os.path.relpath(path, os.getcwd())
+                return os.path.join(document_root, requested_uri)
+
+            # Disable the logging.
+            def log_message(self, format, *args):
```

Comment:
pylint nit: Can you add a `# pylint: disable=redefined-builtin` to ignore the dummy `format` argument?